core/src/main/java/io/grpc/internal/CallCredentialsApplyingTransportFactory.java

```diff
@@ -58,6 +59,12 @@ public ScheduledExecutorService getScheduledExecutorService() {
     return delegate.getScheduledExecutorService();
   }
 
+  @Override
+  public ClientTransportFactory withNewChannelCredential(ChannelCredentials channelCreds) {
+    return new CallCredentialsApplyingTransportFactory(
```

Comment:
I think we also don't want to support it in this factory. Having support here makes it seems easy to accidentally use call credentials improperly, and it shouldn't actually be called today. When we need it, we can add it at that point.

Reply:
Okay, throw UOE now.